assertion/function/analyzer.go

```diff
@@ -293,10 +300,10 @@ func duplicateFullTriggersFromContractedFunctionsToCallers(
 	// return) into all the callers
 	dupTriggers := map[*types.Func][]annotation.FullTrigger{}
 	for ctrtFunc, calls := range callsByCtrtFunc {
-		// TODO: have to introduce r to make NilAway happy. Remove r when NilAway can tolerate
-		//  this.
 		r := funcResults[ctrtFunc]
 		if r == nil {
+			// should not happen since funcResults should contain all the functions including any
+			// contracted functions.
```

Comment:
If we are _certain_ this can't happen due to NilAway invariants... should we panic here instead? (The top level handler will convert that to a reported error rather than a crash, I believe and it has the advantage of making debugging easier if this assumption is ever broken).

Reply:
Included.